From d5a3e749a30613294f41386180aaf31dfb1a9bee Mon Sep 17 00:00:00 2001
From: Nate Graham <nate@kde.org>
Date: Thu, 14 Oct 2021 15:44:33 -0600
Subject: [PATCH] sddm-theme: fix missing password field on "Other" page

On this page there is no concept of the user being passwordless because
no username has been specified by the user yet; we cannot know ahead of
time. As a result, userList.currentItem.needsPassword is blank so the
password field gets hidden.

We should just assume that in username/password entry mode, all users
have a password set. For any users where this is not the case, they can
just leave the password blank, and it's not any worse than it was in
Plasma 5.22.

BUG: 443737
FIXED-IN: 5.23.1


(cherry picked from commit f038ab3f494f26b443b10ff5dc9f156e1ec16341)
---
 sddm-theme/Login.qml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sddm-theme/Login.qml b/sddm-theme/Login.qml
index 8829666f1..6d3d62804 100644
--- a/sddm-theme/Login.qml
+++ b/sddm-theme/Login.qml
@@ -83,7 +83,7 @@ SessionManagementScreen {
                 }
             }
 
-            visible: userList.currentItem.needsPassword
+            visible: root.showUsernamePrompt || userList.currentItem.needsPassword
 
             Keys.onEscapePressed: {
                 mainStack.currentItem.forceActiveFocus();
@@ -119,7 +119,7 @@ SessionManagementScreen {
 
             icon.name: text.length == 0 ? "go-next" : ""
 
-            text: userList.currentItem.needsPassword ? "" : i18n("Log In")
+            text: root.showUsernamePrompt || userList.currentItem.needsPassword ? "" : i18n("Log In")
             onClicked: startLogin();
         }
     }
-- 
2.33.0

